hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -281,15 +277,8 @@ private MemberListSnapshot createSnapshot(int memberListVersion, Collection<Memb
 
         List<MembershipEvent> events = new LinkedList<>();
 
-        // removal events should be added before added events
         for (Member member : deadMembers) {
             events.add(new MembershipEvent(client.getCluster(), member, MembershipEvent.MEMBER_REMOVED, currentMembers));
-            Connection connection = connectionManager.getConnection(member.getUuid());
-            if (connection != null) {
-                connection.close(null,
```

Comment:
did you really need to delete these lines and defer this close to the periodic task even after you used an executor for `fireConnectionRemovedEvent`?

Reply:
No, we don't need to. But when I look at the code after the issue, it made me realize that we are handling this in the wrong place. Notice that this is the only reason for ClusterService to have a reference to ConnectionManager. 

Secondly, if I don't delete the code this path still acquires two locks in the same thread which leaves an open door to making the same mistake in the future.